woocommerce/woocommerce-android
# Issue/6056 interac refund failure event

Closes: #6056 


### Description

This PR adds a tracking event for Interac refund failure. Currently, Interac refund failure can happen in 3 different places.
1. When we fail to fetch the Order. (Interac refund needs to know the order id without which the refund cannot happen)
2. When the `chargeId` on Order is null for whatever reason.
3. When Interac refund fails while `collecting` or `processing` the refund.

### Testing instructions

For scenarios 1 and 2 above, I believe running unit tests and ensuring those pass should be enough. For scenario 3, try to cancel the refund when it is in progress by pressing the back button or using a simulated card reader with some decline code

## Test steps for Scenario 3
1. Create an Order with for a store setup in Canada
2. Collect money for that order via IPP using Interac card
3. Try to refund for the same order using the same Interac card
4. Press back while Interac refund is in progress
5. Check logs and make sure Interac refund failure event is tracked.
